### DISCUSSION TITLE
How can I initialize a schema object without having to set virtuals in Typescript?

### DISCUSSION THREAD

**Discussion by alimoli** [typegoose/typegoose#699] [Q&A]
## Problem

I have a model `User` with the getter `id: string` in order to retrieve the Object ID as string.
When I want to create a new document in the model `User`, I define the properties using the class `User` where my model is defined.
Typescript complains on the fact that I do not pass the virtual getter `id`.
Any suggestion for it? I would like to keep my typed object for the creation.

**User model**
```ts
@modelOptions({
    schemaOptions: {
        timestamps: true,
        toJSON: {virtuals: true},
        toObject: {virtuals: true}
    },
})
export class UserAddress extends TimeStamps {

    _id!: mongoose.Types.ObjectId;

    public get id(): string {
        return this._id.toString()
    }

}

export const UserModel: ReturnModelType<typeof User> = mongoose.models.User || getModelForClass(User)

```

**Creation**
```ts
const user: User = {_id:  new Types.ObjectId()}
await UserModel.create(user)
```

**Compiler Error**
```
TS2741: Property 'id' is missing in type '{ _id: Types.ObjectId; }' but required in type 'User'.
```

## Versions
- Typescript: 9.5.0

**Comment by hasezoey** [comment]
This is a know problem, functions and get/set have to be manually filtered out if you want to use the `Class` type somewhere as a POJO-like interface

i dont know of any non-messy workaround

PS: i will move this question to discussions

Edit: if you dont need to have a manual `id` virtual, mongoose will provide one unless you disable that option
